src/node/routes/vscode.ts

```diff
@@ -33,25 +37,38 @@ export class CodeServerRouteWrapper {
       })
     }
 
-    const { query } = await req.settings.read()
-    if (query) {
-      // Ew means the workspace was closed so clear the last folder/workspace.
-      if (req.query.ew) {
-        delete query.folder
-        delete query.workspace
-      }
+    if (NO_FOLDER_OR_WORKSPACE_QUERY && !WORKSPACE_WAS_CLOSED) {
+      const settings = await req.settings.read()
+      const lastOpened = settings.query || {}
+      // This flag disables the last opened behavior
+      const IGNORE_LAST_OPENED = req.args["ignore-last-opened"]
+      const HAS_LAST_OPENED_FOLDER_OR_WORKSPACE = (!WORKSPACE_WAS_CLOSED && lastOpened.folder) || lastOpened.workspace
```

Comment:
`ew` actually affects both folders and workspaces, I guess we could call it `FOLDER_OR_WORKSPACE_WAS_CLOSED` or something.

```suggestion
      const HAS_LAST_OPENED_FOLDER_OR_WORKSPACE = lastOpened.folder || lastOpened.workspace
```

Reply:
<img width="680" alt="image" src="https://user-images.githubusercontent.com/3806031/156456127-4dc51124-1e84-415b-9cce-f01b907ec343.png">

Is this commit suggestion right? I would assume I only need to change the name?

Reply:
Well you can remove it because it is always true (I edited my comment too late to say that)

Reply:
imo no need to change the name unless you want to